spec/controllers/spree/api/v2/storefront/wished_products_controller_spec.rb

```diff
@@ -0,0 +1,99 @@
+RSpec.describe Spree::Api::V2::Storefront::WishedProductsController, type: :request do
+  let(:wishlist)   { create(:wishlist) }
+  let(:user)       { wishlist.user }
+  let(:product)    { create(:variant) }
+  let(:headers)    { headers_bearer }
+
+  include_context 'API v2 tokens'
+
+  context '#create' do
+
+    it 'must permit add product to the default wishlist' do
+      post "/api/v2/storefront/wishlists/#{wishlist.id}/wished_products", headers: headers, params: {
+        wished_product: {
+          variant_id: product.id
+        }
+      }
+      expect(json['data']['relationships']['variant']['data']['id']).to eq(product.id.to_s)
+    end
+
+    it 'will add the item to list identified by `wishlist_id` if passed' do
+      other_wishlist = create(:wishlist, user: user)
+
+      post "/api/v2/storefront/wishlists/#{other_wishlist.id}/wished_products", headers: headers, params: {
+        wished_product: {
+          variant_id: product.id
+        }
+      }
+      expect(Spree::WishedProduct.count).to eq(1)
+      expect(Spree::WishedProduct.last.wishlist_id).to eq(other_wishlist.id)
+    end
+
+    it 'can not add product if missing variant' do
+      post "/api/v2/storefront/wishlists/#{user.wishlist.id}/wished_products", headers: headers, params: {
+        wished_product: {
+          nodata_id: product.id
+        }
+      }
+      expect(response).to have_http_status(422)
+    end
+
+  end
+
+  context '#update' do
+    let(:bad_user) { create(:user) }
+    let(:new_product) { create(:variant) }
+
+    before do
+      bad_user.generate_spree_api_key!
+      @wished_product = wishlist.wished_products.create( {
+        variant_id: product.id
+      })
+    end
+
+    it 'must permit update wishlist product' do
+      put "/api/v2/storefront/wishlists/#{user.wishlist.id}/wished_products/#{@wished_product.id}", headers: headers, params: {
+        wished_product: {
+          variant_id: new_product.id,
```

Comment:
@chhlga so is this a product Id or variant ID?

Reply:
@damianlegawiec variant. Just variable name. fixed